mmv1/third_party/terraform/transport/config.go.tmpl

```diff
@@ -520,6 +573,24 @@ func (c *Config) LoadAndValidate(ctx context.Context) error {
 	return nil
 }
 
+// getExternalAccountConfig returns an externalaccount.Config based on the Config object.
+// The external account config is intended to be used to create a token source.
+func (c *Config) getExternalAccountConfig(clientScopes []string) externalaccount.Config {
+	eaConfig := externalaccount.Config{
+		Audience:         c.ExternalCredentials.Audience,
+		SubjectTokenType: "urn:ietf:params:oauth:token-type:jwt",
+		ServiceAccountImpersonationURL: fmt.Sprintf("https://iamcredentials.googleapis.com/v1/projects/-/serviceAccounts/%s:generateAccessToken", c.ExternalCredentials.ServiceAccountEmail),
+		Scopes:               clientScopes,
+		SubjectTokenSupplier: c.ExternalCredentials,
+	}
+	// If UniverseDomain is set, the externalaccount package will use it to set the TokenURL (https://sts.UNIVERSE_DOMAIN/v1/token). Otherwise TokenURL defaults to https://sts.googleapis.com/v1/token
+	if c.UniverseDomain != "" {
```

Comment:
In other universe domain checks, we also verify that it is not the default universe like in : https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/transport/config.go.tmpl#L1228

We do this to avoid any weird cases of trying to override the default universe with the default universe - I'm assuming we'd want to be consistent with that check here

Reply:
addressed in https://github.com/GoogleCloudPlatform/magic-modules/pull/12714/commits/983837c8af1aeb0168e30e0303a25af6335ff8ce